src/SLCore.Listeners/Implementation/AnalysisListener.cs

```diff
@@ -29,18 +29,42 @@ namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation;
 [PartCreationPolicy(CreationPolicy.Shared)]
 internal class AnalysisListener : IAnalysisListener
 {
+    private readonly IAnalysisService analysisService;
+    private readonly IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter;
+
+    [ImportingConstructor]
+    public AnalysisListener(IAnalysisService analysisService, IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter)
+    {
+        this.analysisService = analysisService;
+        this.raiseIssueParamsToAnalysisIssueConverter = raiseIssueParamsToAnalysisIssueConverter;
+    }
+
     public Task DidChangeAnalysisReadinessAsync(DidChangeAnalysisReadinessParams parameters)
     {
         return Task.CompletedTask;
     }
 
     public void RaiseIssues(RaiseIssuesParams parameters)
     {
-        throw new NotImplementedException();
+        if (parameters.analysisId.HasValue)
+        {
+            var filepath = parameters.issuesByFileUri.Single().Key.LocalPath;
+
+            var issues = GetSupportedLanguageIssues(raiseIssueParamsToAnalysisIssueConverter.GetAnalysisIssues(parameters));
```

Comment:
You can filter before conversion

Reply:
To do that we also need to change the `raiseIssueParamsToAnalysisIssueConverter`. This change makes sense but I'll open a separate ticket for that.

Reply:
Why? The dto has the same rule key value

Reply:
we pass the params object to the converter I do not want to manipulate the object itself or create a copy of it. It would be a quick change but deserves a separate ticket IMO.